evap/staff/views.py

```diff
@@ -801,12 +814,58 @@ def questionnaire_copy(request, questionnaire_id):
         else:
             return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
     else:
-        questionnaire = get_object_or_404(Questionnaire, id=questionnaire_id)
+        form, formset = get_identical_form_and_formset(questionnaire_id)
+        return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
+
+
+def find_questionnaire_name(old_questionnaire):
+    timestamp = datetime.date.today()
+    new_name_de = '{} (until {})'.format(old_questionnaire.name_de, str(timestamp))
+    new_name_en = '{} (until {})'.format(old_questionnaire.name_en, str(timestamp))
+
+    if Questionnaire.objects.filter(Q(name_de=new_name_de) | Q(name_en=new_name_en)):
+        return None, None
+    return new_name_de, new_name_en
+
+
+@staff_required
+def questionnaire_new_version(request, questionnaire_id):
+    if request.method == "POST":
+        questionnaire = Questionnaire()
         InlineQuestionFormset = inlineformset_factory(Questionnaire, Question, formset=AtLeastOneFormSet, form=QuestionForm, extra=1, exclude=('questionnaire',))
 
-        form = QuestionnaireForm(instance=questionnaire)
-        formset = InlineQuestionFormset(instance=questionnaire, queryset=questionnaire.question_set.all())
+        form = QuestionnaireForm(request.POST, instance=questionnaire)
+        formset = InlineQuestionFormset(request.POST.copy(), instance=questionnaire, save_as_new=True)
```

Comment:
because it is done with the other questionnaire copy view a few lines above. I didn't question it. It was last edited by you a year ago. Do you remember why you copied the POST request, or why it could be necessary?